### ISSUE TITLE
New Job Needed: Aquitanian Heuristic "Pitch" Finding

### ISSUE THREAD

**Issue by martha-thomae** [DDMAL/Rodan#1240]
Summarising the discussion we had with @fujinaga:

1. We realised that the one staff line of Aquitanian notation is not visible in all manuscripts. Some of them have it as a red line, a black line, or a pencil line; but others just have it as if it was some kind of scrape. That is actually the case for the Salamanca Missal we plan on using and doing OMR on, see some folios here: https://pemdatabase.eu/source/48357 (go to musical items and click on any of them). 

   We realised that, at least in this manuscript, the distance between each scrapped line seems to be constant; and that, at least in this manuscript, it seems safe to assume that the “staff line” is always in the midpoint between two lines of text. @annamorphism and @elsinhadl, do you know if this can be generalizable?

   We want to use this idea (detecting the midpoint between two consecutive text lines from the text layer) to detect where this one staff line is located. **First job.**

2. We are thinking that Deanna’s work for finding the location of the neume components with respect to the one staff line (which we will assume has a pitch of F) would be done by creating a new "Heuristic Pitch Finding” Job, but suited for Aquitanian notation. This job will find the distance between the note and the staff line and then, based on the average of those distances, it will determine what is the size of the “step” and determine if the notes are 1, 2, 3, etc. steps up or down the staff line. **Second job (heuristic 'pitch' finding for Aquitanian).**

   We are not calculating any other staff lines based on the one staff line we have, we are just calculating the location of the notes in terms of steps regarding to that staff line, and then converting those to pitches (assuming that the one staff line has a pitch of F). The pitches are calculated just for things to work out in Neon.

   Usually MEI files in Aquitanian notation use `@loc` with integer values to define the position of the notes, but we will use actual pitch information (with `@pname` and `@oct`) so that things work in Neon.

**Comment by annamorphism** [comment]
@martha-thomae I thought we were assuming a pitch of F for the line?
The staff line should indeed be at the midpoint between text lines (modulo a certain amount of human error--but that would also be true for e.g. the distance between staff lines on a four-line staff.) I imagine, though, that the easiest part of the text line to find is the midpoint of the letter height, which is a little bit above the line on which the text is written. So everything would be offset by half a letter-height, if that makes sense...

**Comment by elsinhadl** [comment]
The line could assume different meanings: A, B, D, F, G. It depends by the mode of the chant and the location of the rhombus puncta. 
I believe it is a very good idea to measure the midpoint between two text lines bearing notation to find the musical line. The Salamanca Missal is a good quality manuscript carefully prepared, therefore this system should work!

**Comment by martha-thomae** [comment]
> @martha-thomae I thought we were assuming a pitch of F for the line?

You are absolutely right, @annamorphism, that is what we talked about. I will edit it in the original comment. Thank you!
